backend/src/managers/UserManager.ts

```diff
@@ -659,4 +659,13 @@ export default class UserManager {
             return false;
         }
     }
+
+    async getUsernameSuggestions(startsWith: string): Promise<string[]> {
+        const cached = this.userCache.getUsernameSuggestion(startsWith);
```

Comment:
Note: this variant of the cache can grow without limit. It's possible to exhaust server's RAM by generating queries with random username prefixes.

I'd suggest to either use LRU cache instead of map, or, better, switch to the Trie (see comment above).